packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/io.dart';
+import '../doctor_validator.dart';
+
+class WindowsVersionValidator extends DoctorValidator {
+  /// Validator to be run with `flutter doctor` to check
+  /// Windows host machines if they are running supported versions,
+  /// current unsupported versions = 7, 8
+  const WindowsVersionValidator({required ProcessManager processManager})
+      : _processManager = processManager,
+        super('Windows Version');
+
+  final ProcessManager _processManager;
+
+  @override
+  Future<ValidationResult> validate() async {
+    final ProcessResult result;
+    try {
+      result = _processManager.runSync(<String>['systeminfo']);
+    } on ProcessException {
+      return const ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[],
+        statusInfo:
+            'Unable to run Windows version check using built-in `systeminfo`',
+      );
+    }
+
+    if (result.exitCode != 0) {
+      return const ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[],
+        statusInfo: 'Exit status from running `systeminfo` was unsuccessful',
+      );
+    }
+
+    final String resultStdout = result.stdout as String;
+
+    // Define the major versions that are not supported
+    const List<String> unsupportedVersions = <String>[
+      '7',
+      '8',
+    ];
+
+    final List<String> systemInfoElements = resultStdout.split('\n');
+
+    // Regular expression pattern for identifying
+    // semantic versioned strings
+    // (ie. 10.5.4123)
+    final RegExp regex = RegExp(r'^([0-9]+)\.([0-9]+)\.([0-9]+)$');
+
+    // Define the list that will contain the matches;
+    // if ran successfully, this list should have only
+    // one item
+    final List<String> versionList = <String>[];
+
+    // Use two booleans to identify when you have found
+    // the word 'version' and a version number that matches
+    // the regex pattern above; only once both are found do
+    // we report back a valid version
+    bool versionText = false;
+    bool versionSemver = false;
+    String? version;
+    for (final String curLine in systemInfoElements) {
```

Comment:
Rather than breaking down each line into individual chunks and running the regex over that, use something like String.contains on each line and pull the version info out of the match object returned.

Reply:
That's a good point, I'm wondering if it would be worth ditching looping over each line and use the `multiLine` boolean on `RegExp` for matches. What are your thoughts on that approach?

Reply:
This link should show the pattern working in a demo.

https://regex101.com/r/EA9AAq/1

Using the above demo, the middle group would be our version we're interested in

Reply:
I would not use multiLine unless you potentially want a match that spans multiple lines

Reply:
If you have something that matches your line, you can just run that over the whole string and it should work

Reply:
sounds good, editing the pattern matcher to account for that